waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -0,0 +1,82 @@
+module Wasp.Generator.ServerGenerator.JobGenerator
+  ( genJobs,
+    genJobFactories,
+  )
+where
+
+import Data.Aeson (object, (.=))
+import Data.Maybe
+  ( fromJust,
+  )
+import qualified GHC.Enum as Enum
+import StrongPath
+  ( Dir,
+    File',
+    Path,
+    Path',
+    Posix,
+    Rel,
+    parseRelFile,
+    reldir,
+    reldirP,
+    relfile,
+    (</>),
+  )
+import Wasp.AppSpec (AppSpec, getJobs)
+import Wasp.AppSpec.Job (Job (perform))
+import Wasp.Generator.ExternalCodeGenerator.Common (GeneratedExternalCodeDir)
+import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.JsImport (getJsImportDetailsForExtFnImport)
+import Wasp.Generator.Monad (Generator)
+import Wasp.Generator.ServerGenerator.Common
+  ( ServerSrcDir,
+  )
+import qualified Wasp.Generator.ServerGenerator.Common as C
+
+-- | TODO: Make this not hardcoded!
+relPosixPathFromJobFileToExtSrcDir :: Path Posix (Rel (Dir ServerSrcDir)) (Dir GeneratedExternalCodeDir)
+relPosixPathFromJobFileToExtSrcDir = [reldirP|../ext-src|]
+
+data JobFactory = PassthroughJobFactory
+  deriving (Show, Eq, Ord, Enum, Enum.Bounded)
+
+jobFactories :: [JobFactory]
+jobFactories = enumFrom minBound :: [JobFactory]
+
+-- TODO: In future we will detect what type of JobFactory
+-- to use based on what the Job is using.
+jobFactoryForJob :: Job -> JobFactory
+jobFactoryForJob _ = PassthroughJobFactory
+
+jobFactoryFilePath :: JobFactory -> Path' (Rel d) File'
+jobFactoryFilePath PassthroughJobFactory = [relfile|src/jobs/PassthroughJobFactory.js|]
```

Comment:
One more minor thing I failed to notice during my original review. @Martinsos pointed me to the preferred order of functions in one of my previous PRs: https://github.com/wasp-lang/wasp/pull/500#discussion_r824602425. We should probably apply the same rule here.

Can we perhaps make the linter take care of this for us?

Reply:
Hah I don't think linter can do that :D!
It is not a hard rule, it is mostly about telling a story. So as in a story, you start with more general picture, and then you dive into the details of the plot. The same idea is with the code -> file should start with high level stuff, and as you read it, you get to specific details. Like a newspaper article.

Reply:
In general I totally agree with this principle. Here, I was thinking since they are kind of type level helpers for JobFactory keeping them closer to that declaration made sense, but they can go below their use, np.